### PR TITLE
MGMT-18905: moving to cloud config terraform

### PR DESCRIPTION
**Pull request by eifrach**
Due to some issue with the Nutanix testing flow, we changing the way it works

1. Using Cloud image instead of building one by packer
2. Removing the packer step, cause it's not needed. Temporary, changed  the packer module to  a dummy until release flow will be patched and will be removed in a later date
3. Switching to Rocky linux to align the tests with other flow’s
4. Introducing `cloud-config.yaml` for initial VM configuration ( i.e. cloud-init)

### CODE REVIEW

File: terraform_files/nutanix-ci-machine/variables-nutanix.tf [hunk 1]
```diff
@@ -57,10 +51,32 @@ variable "vcpu" {
 
 variable "disk_size" {
   type        = number
-  default     = 240
+  default     = 650
```
Comment (eliorerz) [inline terraform_files/nutanix-ci-machine/variables-nutanix.tf:54]: Why is it so big? I think that 240 was too much

**Review by eliorerz — CHANGES_REQUESTED** [review]
Nice! Just a few minor comments. Also, are we plan to do that change to vsphere?

File: terraform_files/nutanix-ci-machine/variables-nutanix.tf (hunk 1 above)
Comment (eifrach) [reply terraform_files/nutanix-ci-machine/variables-nutanix.tf:54]: I was too but it seems that the test infra needs a lot more 

log from run before this change
```
{  age nmstate-devel-2.2.35-1.el9.x86_64 needs 410MB more space on the / filesystem
  installing package pipewire-jack-audio-connection-kit-1.0.1-1.el9.x86_64 needs 410MB more space on the / filesystem
  installing package pipewire-alsa-1.0.1-1.el9.x86_64 needs 410MB more space on the / filesystem
  installing package pipewire-pulseaudio-1.0.1-1.el9.x86_64 needs 411MB more space on the / filesystem
  installing package libvirt-client-10.5.0-5.el9.x86_64 needs 412MB more space on the / filesystem
  installing package libvirt-devel-10.5.0-5.el9.x86_64 needs 414MB more space on the / filesystem
  installing package rpm-plugin-selinux-4.16.1.3-34.el9.x86_64 needs 414MB more space on the / filesystem
  installing package gcc-11.5.0-2.el9.x86_64 needs 493MB more space on the / filesystem
  installing package openssl-devel-1:3.2.2-6.el9.x86_64 needs 527MB more space on the / filesystem
  installing package xorriso-1.5.4-4.el9.x86_64 needs 528MB more space on the / filesystem
  installing package postgresql-13.16-1.el9.x86_64 needs 531MB more space on the / filesystem
  installing package sqlite-3.34.1-7.el9.x86_64 needs 533MB more space on the / filesystem

Error Summary
-------------
Disk Requirements:
   At least 533MB more space needed on the / filesystem.
  ```

**Comment by eifrach** [comment]
> Nice! Just a few minor comments. Also, are we plan to do that change to vsphere?

Yes, maybe we can join the flows.

we may move the cloud config to a different module and  pull it in Nutanix/vSphere etc ...

**Comment by eifrach** [comment]
> Nice! Just a few minor comments. Also, are we plan to do that change to vsphere?

Yes, maybe we can join the flows.

we may move the cloud config to a different module and  pull it in Nutanix/vSphere etc ...

File: terraform_files/nutanix-ci-machine/variables-nutanix.tf (hunk 1 above)
Comment (eliorerz) [reply terraform_files/nutanix-ci-machine/variables-nutanix.tf:54]: The description says that this field is in GB, the error message is in MB.
Am I missing somehing?

Comment (eifrach) [reply terraform_files/nutanix-ci-machine/variables-nutanix.tf:54]: yeah I wiill  update it you correct

Comment (eifrach) [reply terraform_files/nutanix-ci-machine/variables-nutanix.tf:54]: I've checked - this is reported  in GB in the var and converted to bytes in the terrraform file

File: terraform_files/nutanix-ci-machine/variables-nutanix.tf [hunk 2]
```diff
@@ -57,10 +51,32 @@ variable "vcpu" {
...
+variable "ssh_private_key" {
+  type = string
+  description = "The private ssh key path, used to authenticate against the new template"
+  sensitive   = true
+}
+
+variable "cloud_config_file" {
+  type = string
+  default = "cloud-config.yaml"
+  description = "Cloud config file path"
```
[the remark this answers is not in the document]
Comment (eifrach) [reply terraform_files/nutanix-ci-machine/variables-nutanix.tf]: done

File: terraform_files/nutanix-ci-machine/main.tf [hunk 3]
```diff
@@ -35,6 +76,7 @@ resource "nutanix_virtual_machine" "vm" {
   num_vcpus_per_socket = var.cores_per_socket
   memory_size_mib      = var.memory
   num_sockets          = var.vcpu
+  guest_customization_cloud_init_user_data = "${data.local_file.cloud_config_file.content_base64}"
```
Comment (adriengentil) [inline terraform_files/nutanix-ci-machine/main.tf:79]:
```suggestion
  locals {
   user_data = <<EOF
#cloud-config
...
EOF
   }
  guest_customization_cloud_init_user_data = base64encode(local.user_data)
```

wouldn't it be simpler?

Comment (eifrach) [reply terraform_files/nutanix-ci-machine/main.tf:79]: it  is, and  will work - but, I think we should move the cloud config creation to seperated module so we can imported it to others. 

so we will have one config setup for all CI deployment. 

maybe we can play with this config on the next step

**Review by adriengentil — APPROVED** [review]

**Pull request merged**